src/main/kotlin/uk/gov/justice/digital/hmpps/courtregister/config/SecurityUserContext.kt

```diff
@@ -14,5 +14,11 @@ class SecurityUserContext {
     }
 
   val principal: String
-    get() = authentication?.principal!!
+    get() {
+      return if (authentication?.principal != null) {
```

Comment:
You could do `get() = authentication?.principal?:"anonymous"`.  I think so anyway.